awx/main/tests/unit/test_indirect_query_discovery.py

```
@@ -416,6 +417,62 @@ def test_queries_not_collected_when_option_disabled(self, mock_fallback, mock_fi
 
+class TestListCollectionsFailureResilience:
+    """list_collections() reuses ansible-galaxy CLI internals (with_collection_artifacts_manager)
+    that read CLI-only context.CLIARGS entries directly. Since AAP-74343 this callback always
+    runs (previously it only ran when FEATURE_INDIRECT_NODE_COUNTING_ENABLED was on), so any
+    latent fragility in those internals now affects every job, not just opted-in ones. A failure
+    here should degrade gracefully instead of aborting the whole callback and losing
+    ansible_version/installed_collections for the job.
+    """
+
+    @mock.patch('awx.playbooks.library.indirect_instance_count.list_collections')
+    @mock.patch.dict('os.environ', {'AWX_ISOLATED_DATA_DIR': '/tmp/artifacts'})
```

Ruff (0.16.1)

[error] Probable insecure usage of temporary file or directory: "/tmp/artifacts"

(S108)